bot/bot.py

```diff
@@ -44,9 +45,23 @@ def __init__(self, *args, redis_session: RedisSession, **kwargs):
             # Since statsd is UDP, there are no errors for sending to a down port.
             # For this reason, setting the statsd host to 127.0.0.1 for development
             # will effectively disable stats.
-            statsd_url = "127.0.0.1"
+            statsd_url = LOCALHOST
 
-        self.stats = AsyncStatsClient(self.loop, statsd_url, 8125, prefix="bot")
+        self.stats = AsyncStatsClient(self.loop, LOCALHOST)
+        self.connect_statsd(statsd_url)
+
+    def connect_statsd(self, statsd_url: str, retry_after: int = 2, attempt: int = 1) -> None:
+        """Callback used to retry a connection to statsd if it should fail."""
+        if attempt > 5:
+            log.error("Reached 10 attempts trying to reconnect AsyncStatsClient. Aborting")
```

Comment:
This error message has the wrong count. As things currently stand, it should say "6 attempts".